framework/snippets/src/interactor_result.rs

```diff
@@ -49,34 +50,70 @@ impl<T: TopDecodeMulti> InteractorResult<T> {
         }
     }
 
+    // Returns the address of the newly deployed smart contract.
     pub fn new_deployed_address(&self) -> Address {
-        let event = self
-            .find_log(LOG_IDENTIFIER_SC_DEPLOY)
-            .expect("SCDeploy event log not found");
-        let topics = event.topics.as_ref().expect("missing topics");
-        assert_eq!(topics.len(), 2, "`SCDeploy` is expected to have 2 topics");
+        self.handle_signal_error_event();
+
+        let topics = self.handle_sc_deploy_event();
         let address_raw = base64::decode(topics.get(0).unwrap()).unwrap();
         let address = Address::from_slice(address_raw.as_slice());
+
         info!("new address: {}", bech32::encode(&address));
         address
     }
 
+    // Returns the token identifier of the newly issued non-fungible token.
     pub fn issue_non_fungible_new_token_identifier(&self) -> String {
+        self.handle_signal_error_event();
+
         let second_scr = self
             .scrs
             .iter()
             .find(|scr| scr.data.starts_with("@00@"))
-            .expect("no token identifier SCR found");
-
-        // TODO: error handling
-        let mut split = second_scr.data.split('@');
-        let _ = split.next().unwrap();
-        let _ = split.next().unwrap();
-        let encoded_tid = split
-            .next()
+            .unwrap_or_else(|| panic!("no token identifier SCR found"));
+        let encoded_tid = second_scr
+            .data
+            .split('@')
+            .nth(3)
             .unwrap_or_else(|| panic!("bad issue token SCR data: {}", second_scr.data));
+
         String::from_utf8(hex::decode(encoded_tid).unwrap()).unwrap()
     }
+
+    // Handles a signalError event, if present.
+    fn handle_signal_error_event(&self) {
+        if let Some(event) = self.find_log(LOG_IDENTIFIER_SIGNAL_ERROR) {
+            let topics = event
+                .topics
+                .as_ref()
+                .unwrap_or_else(|| panic!("missing topics"));
+            assert_eq!(
+                topics.len(),
+                2,
+                "`{LOG_IDENTIFIER_SIGNAL_ERROR}` is expected to have 2 topics"
+            );
+            let error_raw = base64::decode(topics.get(1).unwrap()).unwrap();
+            let error = String::from_utf8(error_raw).unwrap();
+            panic!("error: {error:#?}");
```

Comment:
In this case we should actually return a Result. In some cases we are explicitly verifying that a tx should fail, and don't want out program to crash.

It's probably best to build some custom error type, or at least an alias for String.

We need:
```
fn handle_signal_error_event(&self) -> Result<(), TxError> ...
pub fn value(&self) -> Result<T, TxError> { ...
pub fn new_deployed_address(&self) ->  Result<Address, TxError>
```
etc.